### DISCUSSION TITLE
How best to use as library in Android (Kotlin/Java)?

### DISCUSSION THREAD

**Discussion by AltNico** [tulir/whatsmeow#468] [General]
Hey there!

I'm currently wondering how I could best make use of this wonderful library on Android, or rather, calling the library's functions from Java and Kotlin.

I already successfully cross-compiled the `mdtest` binary for `aarch64` and executed it as part of an Android application, piping a `pair-phone +xxxx` command into its input and reading out the resulting linking code. By this I could confirm that whatsmeow also works natively under Android, as I could see messages sent to the linked WhatsApp account in the Android application's log files.

Reading articles like [Calling Go Functions from Other Languages](https://medium.com/learning-the-go-programming-language/calling-go-functions-from-other-languages-4c7d8bcc69bf) and others gives me the impression that maybe writing a wrapper application in Go would be the way to go which would expose a C-native interface which I could then call from Android/Kotlin.

What do more experienced Go developers think about this? Should I drop this idea and directly use [WuzAPI](https://github.com/asternic/wuzapi)?

Thanks for your help!

**Comment by meucontato-developer** [comment]
Is anyone in 2025 looking for a way to use this library on Android?
